src/losses/functions.jl

```diff
@@ -46,31 +46,51 @@ function huber_loss(ŷ, y; agg=mean, δ=ofeltype(ŷ, 1))
    agg(((abs_error.^2) .* temp) .* x .+ δ*(abs_error .- x*δ) .* (1 .- temp))
 end
 
+function apply_label_smoothing(y::AbstractArray, label_smoothing)
+    if (label_smoothing < 0 || label_smoothing > 1)
+        throw(ArgumentError("`label_smoothing` must be between 0 and 1"))
+    elseif !iszero(label_smoothing)
+        y = y .* (1 - label_smoothing) .+ label_smoothing* 1 // size(y, 1)
+    end
+    return y
+end
+
+function apply_label_smoothing(y::Number, label_smoothing)
+    if (label_smoothing < 0 || label_smoothing > 1)
+        throw(ArgumentError("`label_smoothing` must be between 0 and 1"))
```

Comment:
I am always tempted by:
```suggestion
        throw(DomainError(label_smoothing, "`label_smoothing` must be between 0 and 1"))
```
but don't super-mind